tests/common-tests.h

```diff
@@ -1511,6 +1521,78 @@ TEST_CASE_METHOD(common_tests, "Use type conversion", "[core][use]")
         CHECK(str == "Hello SOCI!");
     }
 
+    SECTION("int8_t")
+    {
+        int8_t s = 123;
+        sql << "insert into soci_test(id) values(:id)", use(s);
+
+        int8_t s2 = 0;
+        sql << "select id from soci_test", into(s2);
+
+        CHECK(s2 == 123);
+    }
+
+    SECTION("int8_t min")
```

Comment:
Is it really worth having 3 sections for each type? I think we could combine all the tests for the same type in a single section, which would make the test slightly shorter and faster to run.

Also, I'd really like to know what happens when trying to insert an out of range value into a column too small (or "too signed") to contain it, it would be great to add some tests checking that this behaves as expected too.

Reply:
> Is it really worth having 3 sections for each type? I think we could combine all the tests for the same type in a single section, which would make the test slightly shorter and faster to run.

I don't really care if it's multiple small sections or a single bigger one, so we can combine that, of course.

> Also, I'd really like to know what happens when trying to insert an out of range value into a column too small (or "too signed") to contain it, it would be great to add some tests checking that this behaves as expected too.

See above, it's not really a problem anymore. But I added the checks where necessary.